pipelinerl/finetune/rl/__init__.py

```diff
@@ -260,37 +263,78 @@ def rl_step(
     )
 
     approx_kl = torch.exp(log_ratio_ref_new_clamp) - log_ratio_ref_new_clamp - 1  # Schulman KL approx
+    approx_kl_new_old = torch.exp(log_ratio_new_old) - log_ratio_new_old - 1  # Schulman KL approx
 
     assert torch.isfinite(approx_kl).all(), f"approx_kl is not finite: {approx_kl}"
     entropy_bonus_coef = linear_decay_coef(current_step, max_step, config.entropy_bonus, config.final_entropy_bonus)
     kl_coef = linear_decay_coef(current_step, max_step, config.kl_coef, config.final_kl_coef)
 
     # compute algorithm-specific losses
+    policy_loss_total = None
     match config.policy_loss:
         case "ppo":
             surr1 = ratio_new_old * log_p_weights
-            clamped_ratio = torch.clamp(ratio_new_old, 1 - config.epsilon, 1 + config.epsilon)
+            assert config.epsilon_low == config.epsilon_high, "PPO requires epsilon_low == epsilon_high"
```

Comment:
What's the reason for this? We wouldn't be able to clamp asymmetrically like in DAPO for GRPO.

Reply:
the reason for this is to keep the old functionality, epsilon needs to be symmetrical for PPO